### DISCUSSION TITLE
dpkg-preconfigure: unable to re-open stdin:

### DISCUSSION THREAD

**Discussion by dinomoss** [actions/virtual-environments#5096] [Installed software]
### Description

I have added env       **DEBIAN_FRONTEND: noninteractive**    but still have the same issue on multiple places. 

  - task: PowerShell@2
    name: buildVm
    displayName: 'Build VM'
    inputs:
      targetType: filePath
      filePath: ./scripts/build-image.ps1
      arguments: -ResourcesNamePrefix $(setVars.ResourcesNamePrefix) `
                        -ClientId $(CLIENT_ID) `
                        -ClientSecret $(CLIENT_SECRET) `
                        -TemplatePath $(TemplatePath) `
                        -ResourceGroup $(AZURE_RESOURCE_GROUP) `
                        -StorageAccount $(AZURE_STORAGE_ACCOUNT) `
                        -SubscriptionId $(AZURE_SUBSCRIPTION) `
                        -TenantId $(AZURE_TENANT) `
                        -Location $(AZURE_LOCATION) `
                        -GitHubFeedToken $(GITHUB_TOKEN)
    env:
      PACKER_LOG: 1
      PACKER_LOG_PATH: $(Build.ArtifactStagingDirectory)/packer-log.txt
      **DEBIAN_FRONTEND: noninteractive**

    azure-arm: The following NEW packages will be installed:
    azure-arm:   powershell
    azure-arm: 0 upgraded, 1 newly installed, 0 to remove and 0 not upgraded.
    azure-arm: Need to get 69.7 MB of archives.
    azure-arm: After this operation, 193 MB of additional disk space will be used.
    azure-arm: Get:1 https://packages.microsoft.com/ubuntu/20.04/prod focal/main amd64 powershell amd64 7.2.1-1.deb [69.7 MB]
    azure-arm: Fetched 69.7 MB in 2s (40.7 MB/s)
    azure-arm: Selecting previously unselected package powershell.
    azure-arm: (Reading database ... 59975 files and directories currently installed.)
    azure-arm: Preparing to unpack .../powershell_7.2.1-1.deb_amd64.deb ...
    azure-arm: Unpacking powershell (7.2.1-1.deb) ...
    azure-arm: Setting up powershell (7.2.1-1.deb) ...
    azure-arm: Processing triggers for man-db (2.9.1-1) ...
**==> azure-arm: debconf: unable to initialize frontend: Dialog
==> azure-arm: debconf: (Dialog frontend will not work on a dumb terminal, an emacs shell buffer, or without a controlling terminal.)
==> azure-arm: debconf: falling back to frontend: Readline
==> azure-arm: debconf: unable to initialize frontend: Readline
==> azure-arm: debconf: (This frontend requires a controlling tty.)
==> azure-arm: debconf: falling back to frontend: Teletype
==> azure-arm: dpkg-preconfigure: unable to re-open stdin:**

### Virtual environments affected

- [ ] Ubuntu 18.04
- [X] Ubuntu 20.04
- [ ] macOS 10.15
- [ ] macOS 11
- [ ] Windows Server 2016
- [ ] Windows Server 2019
- [ ] Windows Server 2022

### Image version and build link

    azure-arm: After this operation, 26.1 MB of additional disk space will be used.
    azure-arm: Get:1 https://packages.microsoft.com/ubuntu/20.04/prod focal/main amd64 moby-compose amd64 2.2.3+azure-1 [6318 kB]
    azure-arm: Fetched 6318 kB in 0s (32.8 MB/s)
    azure-arm: Selecting previously unselected package moby-compose.
    azure-arm: (Reading database ... 60916 files and directories currently installed.)
    azure-arm: Preparing to unpack .../moby-compose_2.2.3+azure-1_amd64.deb ...
    azure-arm: Unpacking moby-compose (2.2.3+azure-1) ...
    azure-arm: Setting up moby-compose (2.2.3+azure-1) ...
**==> azure-arm: debconf: unable to initialize frontend: Dialog
==> azure-arm: debconf: (Dialog frontend will not work on a dumb terminal, an emacs shell buffer, or without a controlling terminal.)
==> azure-arm: debconf: falling back to frontend: Readline
==> azure-arm: debconf: unable to initialize frontend: Readline
==> azure-arm: debconf: (This frontend requires a controlling tty.)
==> azure-arm: debconf: falling back to frontend: Teletype
==> azure-arm: dpkg-preconfigure: unable to re-open stdin:**
    azure-arm: Client:
    azure-arm:  Context:    default
    azure-arm:  Debug Mode: false
    azure-arm:  Plugins:
    azure-arm:   buildx: Docker Buildx (Docker Inc., 0.7.1+azure-2)
    azure-arm:   compose: Docker Compose (Docker Inc., 2.2.3+azure-1)
    azure-arm:

### Repro steps

This is my pipeline: 

jobs:
- job: imagegeneration
  displayName: Image Generation (${{ parameters.image_type }})
  timeoutInMinutes: 600
  cancelTimeoutInMinutes: 30
  pool: 
    name: ${{ parameters.agent_pool }}

  steps:    
  - script: |
      git clone -b ubuntu20/20220131.1 https://github.com/actions/virtual-environments.git
    displayName: 'Clone the GitHub repository'
  - task: PowerShell@2
    name: setVars
    displayName: 'Set image template variables'
    inputs:
      targetType: 'inline'
      script: |
        $ResourcesNamePrefix = "$(Build.BuildId)"
        $ImageType = "${{ parameters.image_type }}"
        $TemplateDirectoryName = if ($ImageType.StartsWith("ubuntu")) { "linux" } else { "win" }
        $OsType = if ($ImageType.StartsWith("ubuntu")) { "Linux" } else { "Windows" }
        $TemplateDirectoryPath = Join-Path (Join-Path "virtual-environments" "images") $TemplateDirectoryName | Resolve-Path
        $TemplatePath = Join-Path $TemplateDirectoryPath "$ImageType.json"
        $ImageReadmeNameConcat = "${{ parameters.image_type }}-Readme.md"
        $ImageReadmeName = $ImageReadmeNameConcat.Substring(0,1).ToUpper()+$ImageReadmeNameConcat.Substring(1)
        Write-Host "##vso[task.setvariable variable=TemplateDirectoryPath;]$TemplateDirectoryPath"
        Write-Host "##vso[task.setvariable variable=TemplatePath;]$TemplatePath"
        Write-Host "##vso[task.setvariable variable=OsType;isOutput=true;]$OsType"
        Write-Host "##vso[task.setvariable variable=ImageType;isOutput=true;]$ImageType"
        Write-Host "##vso[task.setvariable variable=ResourcesNamePrefix;isOutput=true;]$ResourcesNamePrefix"
        Write-Host "##vso[task.setvariable variable=ImageReadmeName;isOutput=true;]$ImageReadmeName"
  
  - task: PowerShell@2
    name: buildVm
    displayName: 'Build VM'
    inputs:
      targetType: filePath
      filePath: ./scripts/build-image.ps1
      arguments: -ResourcesNamePrefix $(setVars.ResourcesNamePrefix) `
                        -ClientId $(CLIENT_ID) `
                        -ClientSecret $(CLIENT_SECRET) `
                        -TemplatePath $(TemplatePath) `
                        -ResourceGroup $(AZURE_RESOURCE_GROUP) `
                        -StorageAccount $(AZURE_STORAGE_ACCOUNT) `
                        -SubscriptionId $(AZURE_SUBSCRIPTION) `
                        -TenantId $(AZURE_TENANT) `
                        -Location $(AZURE_LOCATION) `
                        -GitHubFeedToken $(GITHUB_TOKEN)
    env:
      PACKER_LOG: 1
      PACKER_LOG_PATH: $(Build.ArtifactStagingDirectory)/packer-log.txt
      DEBIAN_FRONTEND: noninteractive

**Comment by al-cheb** [comment]
It's not an issue, as you can see, we have the same lines in the log and build was successful - https://github.visualstudio.com/virtual-environments/_build/results?buildId=122721&view=logs&j=011e1ec8-6569-5e69-4f06-baf193d1351e&t=5431112d-2b61-5a5f-7042-ef698f761043

```
==> azure-arm: debconf: unable to initialize frontend: Dialog
==> azure-arm: debconf: (Dialog frontend will not work on a dumb terminal, an emacs shell buffer, or without a controlling terminal.)
==> azure-arm: debconf: falling back to frontend: Readline
==> azure-arm: debconf: unable to initialize frontend: Readline
==> azure-arm: debconf: (This frontend requires a controlling tty.)
==> azure-arm: debconf: falling back to frontend: Teletype
==> azure-arm: dpkg-preconfigure: unable to re-open stdin: 
```

**Comment by samus-aran** [comment]
We're locking this discussion because it has not had recent activity and/or other members have asked for more information to assist you but received no response. Thank you for helping us maintain a productive and tidy community for all our members.
